src/panels/lovelace/card-features/hui-select-options-card-feature.ts

```diff
@@ -75,6 +75,14 @@ class HuiSelectOptionsCardFeature
           oldHass?.formatEntityAttributeValue
       ) {
         this._haSelect.layoutOptions();
+        if (this.stateObj) {
+          const newIdx = this.stateObj.attributes.options.findIndex(
+            (option) => option === this.stateObj!.state
+          );
+          if (newIdx >= 0) {
+            this._haSelect.select(newIdx);
```

Comment:
Can we instead use `selectByValue` so we dont have to do the index stuff everywhere?

Reply:
Might be nice but lint says:

> Property 'selectByValue' is protected and only accessible within class 'SelectBase' and its subclasses.ts(2445)

?

Reply:
I'm kinda wondering if this should be pushed down into the class that extends SelectBase instead of in the parent. 

Maybe we can add options as a property to HaSelect, and then in HaSelect, look for options to change and call layout and selectByValue. 

Not sure if there's anything weird about adding options as a property, given that we're not currently doing so.

Reply:
Hmm that seems to work and is much cleaner. I pushed a change with that new approach.